File: algorithms/0217-contains-duplicate/solution.py
# Contains Duplicate - set membership; return True on first repeat.
# Time O(n), Space O(n).

class Solution:
    def containsDuplicate(self, nums: list[int]) -> bool:
        seen: set[int] = set()
        for num in nums:
            if num in seen:
                return True
            seen.add(num)
        return False

File: algorithms/0217-contains-duplicate/Solution.kt
// Contains Duplicate - HashSet membership; add() returns false on collision.
// Time O(n), Space O(n).

class Solution {
    fun containsDuplicate(nums: IntArray): Boolean {
        val seen = HashSet<Int>()
        for (num in nums) {
            if (!seen.add(num)) {
                return true
            }
        }
        return false
    }
}
